app/src/main/java/org/mozilla/focus/gecko/GeckoViewPrompt.java

```diff
@@ -0,0 +1,786 @@
+            if (maxDate != null) {
+                picker.setMaxDate(maxDate.getTime());
+            }
+            datePicker = picker;
+        } else {
+            datePicker = null;
+        }
+
+        final TimePicker timePicker;
+        if (prompt.type == DateTimePrompt.Type.TIME || prompt.type == DateTimePrompt.Type.DATETIME_LOCAL) {
+            final int resId = builder.getContext().getResources().getIdentifier(
+                    "time_picker_dialog", "layout", "android");
+            TimePicker picker = null;
+            if (resId != 0) {
+                try {
+                    picker = (TimePicker) inflater.inflate(resId, /* root */ null);
+                } catch (final ClassCastException | InflateException e) {
+                    Log.e(LOGTAG, "Unable to inflate TimePicker:\n " + e.toString());
+                }
+            }
+            if (picker == null) {
+                picker = new TimePicker(builder.getContext());
+            }
+            setTimePickerTime(picker, cal);
+            picker.setIs24HourView(DateFormat.is24HourFormat(builder.getContext()));
+            timePicker = picker;
+        } else {
+            timePicker = null;
+        }
+
+        final LinearLayout container = addStandardLayout(builder, prompt.title, /* msg */ null);
+        container.setPadding(/* left */ 0, /* top */ 0, /* right */ 0, /* bottom */ 0);
+        if (datePicker != null) {
+            container.addView(datePicker);
+        }
+        if (timePicker != null) {
+            container.addView(timePicker);
+        }
+
+        GeckoResult<PromptResponse> res = new GeckoResult<>();
+
+        final DialogInterface.OnClickListener listener = (dialog, which) -> {
+            if (which == DialogInterface.BUTTON_NEUTRAL) {
+                // Clear
+                res.complete(prompt.confirm(""));
+                return;
+            }
+            if (datePicker != null) {
+                cal.set(datePicker.getYear(), datePicker.getMonth(),
+                        datePicker.getDayOfMonth());
+            }
+            if (timePicker != null) {
+                setCalendarTime(cal, timePicker);
+            }
+            res.complete(prompt.confirm(formatter.format(cal.getTime())));
+        };
+        builder.setNegativeButton(android.R.string.cancel, /* listener */ null)
+                .setNeutralButton(R.string.gv_prompt_clear, listener)
+                .setPositiveButton(android.R.string.ok, listener);
+        createStandardDialog(builder, prompt, res).show();
+
+        return res;
+    }
+
+    @Override
+    @TargetApi(19)
+    public GeckoResult<PromptResponse> onFilePrompt(@NotNull GeckoSession session,
+                                                    @NotNull FilePrompt prompt) {
+        return GeckoResult.fromValue(prompt.dismiss());
+
+        // Disabled file prompt support since we don't see to have supported it before,
+        // and this implementation requires some re-working
+
```

Comment:
Let's remove the commented code. Instead, include a comment pointing to an issue that a contributor can pick up.